src/guards/isGoogleUser.ts

```diff
@@ -0,0 +1,2 @@
+export const isGoogleUser = (user: User | GoogleUser): user is GoogleUser =>
+	(user as GoogleUser)?.googleId !== undefined;
```

Comment:
Personally, instead of casting the user variable to the GoogleUser interface and relying on type assertion, I would prefer simply checking if the user is an instance of GoogleUser. 

export const isGoogleUser = (user: User | GoogleUser): user is GoogleUser =>
    user instanceof GoogleUser && user.googleId !== undefined;